colibri/src/coingecko.rs

```diff
@@ -54,6 +58,50 @@ impl Coingecko {
             Err(e) => error!("Failed to query coingecko for {} due to {}", asset_id, e),
         }
 
+        debug!("Icon not found in coingecko for {}", asset_id);
         None
     }
 }
+
+#[cfg(test)]
+mod test {
+    use crate::create_globaldb;
+    use axum::body::Bytes;
+    use std::sync::Arc;
+
+    use super::Coingecko;
+
+    #[tokio::test]
+    async fn test_coingecko_query() {
+        let globaldb = create_globaldb!().await;
+        let mut server = mockito::Server::new_async().await;
+
+        let coingecko = Coingecko::new(
+            Arc::new(globaldb.expect("Failed to create globaldb for coingecko")),
+            server.url(),
+        );
+        let json = format!(
+            r#"{{"image": {{"small": "{}/coins/images/279/thumb/ethereum.png"}}}}"#,
+            server.url()
+        );
+
+        // mock successful query
+        server
+            .mock("GET", format!("/api/v3/coins/ethereum").as_str())
```

Comment:
shouldn't we have one for HEAD too?

Reply:
this one is not testing the api but the logic for `query_asset_image`. I need to add integration tests for the API but don't want to keep this PR for longer